notebooks/uvis_pam_corrections/WFC3 UVIS Pixel Area Map Corrections.ipynb

```diff
@@ -0,0 +1,733 @@
+    "print(\"Destination path for the UVIS2 PAM:\", dest_path2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we have our WFC3/UVIS PAM files in our working directory."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4.2 Perform PAM Corrections <a id=\"correct\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "We search the science header (scihdr) for the 'CCDCHIP' keyword to determine whether our subarray is on UVIS1 or UVIS2 (see WFC3 UVIS Subarray Diagram // [figure 6.2 in the WFC Instrument Handbook](https://hst-docs.stsci.edu/wfc3ihb/chapter-6-uvis-imaging-with-wfc3/6-4-uvis-field-geometry)).\n",
+    "\n",
+    "The PAM corrected data would be the product of the original data and the pixel area map (cut to the correct pixel dimensions). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if scihdr_512['CCDCHIP'] == 1:\n",
+    "        pam = fits.getdata('UVIS1wfc3_map.fits')\n",
+    "        pamcorr_data_512 = data_512 * pam[y0:y1, x0:x1]\n",
+    "\n",
+    "elif scihdr_512['CCDCHIP'] == 2:\n",
+    "        pam = fits.getdata('UVIS2wfc3_map.fits')\n",
+    "        pamcorr_data_512= data_512 * pam[y0:y1, x0:x1]\n",
+    "else:\n",
+    "        raise Exception('Chip case not handled.')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now have pamcorr_data, our pixel area map corrected FLT image data! We can see the difference between the original image and tha pamcorr image below. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "diff_data_512=(pamcorr_data_512-data_512)\n",
+    "\n",
+    "\n",
+    "#We use zcale to find the min and max for plotting\n",
+    "vmin_diff_512, vmax_diff_512 = zscale(diff_data_512)\n",
+    "\n",
+    "plt.imshow(diff_data_512,vmin=vmin_diff_512,vmax=vmax_diff_512, origin='lower')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Putting It All Together <a id=\"function\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have worked through applying PAM corrections step by step, we apply these corrections to our next two subarrays using a function combining all the steps. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def make_PAMcorr_image_UVIS(data, scihdr, pamdir):\n",
+    "    \"\"\"\n",
+    "    Corrects the geometric distortion of the input image\n",
+    "    data by multiplying by the correct UVIS PAM.\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    data : array\n",
+    "        ****Name of the FITS file.\n",
```

Comment:
The definition should be "Image data before correction." since it's not the name of the fits file.